authors/apps/articles/serializers.py

```diff
@@ -5,19 +5,60 @@
 from rest_framework.pagination import PageNumberPagination
 from collections import OrderedDict
 from rest_framework.response import Response
+from .exceptions import ArticleNotFound
+from authors.apps.utils.baseserializer import BaseSerializer
 
 
-class ArticleSerializer(serializers.ModelSerializer):
+class ArticleSerializer(BaseSerializer):
     """
     Articles serializer
     """
+
+    def __init__(self, *args, **kwargs):
+        super(ArticleSerializer, self).__init__(*args, **kwargs)
+
     author = serializers.ReadOnlyField(source="get_author_details")
 
+    def get_likes_data(self):
+        article = self.context.get('article', None)
+        article = Article.objects.filter(slug=article).first()
```

Comment:
The article can also be accessed from `self.instance`  property of the serializers. This would prevent you from making another query here.

Reply:
https://www.django-rest-framework.org/api-guide/serializers/#accessing-the-initial-data-and-instance